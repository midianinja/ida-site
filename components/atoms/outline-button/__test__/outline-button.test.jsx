import React from 'react';
import { render, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import OutlineButton from '../outline-button';

describe('OutlineButton', () => {
  afterEach(cleanup);

  it('renders without crashing', async () => {
    const { getByTestId } = render(<OutlineButton onClick={() => {}}>Test!</OutlineButton>);

    expect(getByTestId('button')).toHaveTextContent('Test!');
  });

  it('matches snapshot default button', () => {
    const tree = renderer
      .create(<OutlineButton onClick={() => {}}>Default Test!</OutlineButton>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('click', () => {
    const mockOnClick = jest.fn();
    const { getByTestId } = render(<OutlineButton onClick={mockOnClick}>Test!</OutlineButton>);
    
    userEvent.click(getByTestId('button'));
    expect(mockOnClick).toBeCalledTimes(1);
  });

  it('matches snapshot white button', () => {
    const tree = renderer
      .create(
        <OutlineButton
          color="white"
          onClick={() => {}}
        >
            White Test!
        </OutlineButton>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot small button', () => {
    const tree = renderer
      .create(
        <OutlineButton
          small
          onClick={() => {}}
        >
            Small Test!
        </OutlineButton>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
