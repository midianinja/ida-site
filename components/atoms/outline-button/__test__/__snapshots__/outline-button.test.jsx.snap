// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineButton matches snapshot default button 1`] = `
<button
  className="button"
  data-testid="button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Default Test!
</button>
`;

exports[`OutlineButton matches snapshot small button 1`] = `
<button
  className="button __small"
  data-testid="button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Small Test!
</button>
`;

exports[`OutlineButton matches snapshot white button 1`] = `
<button
  className="button __white"
  data-testid="button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  White Test!
</button>
`;
